glmmTMB/R/methods.R

```diff
@@ -637,19 +652,33 @@ residuals.glmmTMB <- function(object, type=c("response", "pearson", "working"),
         mr <- as.numeric(as.numeric(mr)>1)
         names(mr) <- nn  ## restore stripped names
     }
-    r <- mr - fitted(object)
+    mu <- fitted(object)
+    r <- mr - mu
+    fam <- family(object)
     res <- switch(type,
            response=r,
            working = {
-               mu.eta <- family(object)$mu.eta
+               mu.eta <- fam$mu.eta
                p <- predict(object, type = "link", fast = TRUE)
                r/mu.eta(p)
            },
+           deviance = {
+               if (is.null(dr <- fam$dev.resids)) {
+                   warning(
+                       warningCondition(paste0("deviance residuals undefined for family ",
+                                              sQuote(fam$family),": returning NA"),
+                                        class = c("dev_resids_undefined", "glmmTMB_warn")))
+                   return(rep(NA_real_, length(r)))
+               }
+               d.res <- sqrt(pmax(dr(mr, mu, wts), 0))
```

Comment:
In the `nbinom2` case, `dr <- fam$dev.resids` expects 4 arguments, not 3; `theta` is missing.
```
fam$dev.resids
function (y, mu, wt, theta) 
{
    get_nbinom_disp(theta, ".Theta", "theta")
    return(2 * wt * (y * log(pmax(1, y)/mu) - (y + theta) * log((y + 
        theta)/(mu + theta))))
}
```

For example, 
```
> m2=glmmTMB(count~spp, family=nbinom2(), data=Salamanders)
> d2=residuals(m2, type="deviance")
Error in get_nbinom_disp(theta, ".Theta", "theta") : 
  argument "theta" is missing, with no default
```

Reply:
Fixed, thanks for the heads-up (I needed a default `NULL` in the argument list).  This works now too:

```r
m3 <- MASS::glm.nb(count~spp,data = Salamanders)
d3 = residuals(m3, type = "deviance")
all.equal(d2, d3, tolerance = 1e-5)
```

Reply:
Thanks for that. Looks good now :)